clients/fides-js/src/services/external/preferences.ts

```diff
@@ -0,0 +1,58 @@
+import {
+  FidesConfig,
+  FidesOptions,
+  GetPreferencesFnResp,
+  PrivacyExperience,
+} from "../../lib/consent-types";
+import { debugLog } from "../../lib/consent-utils";
+import { CookieKeyConsent } from "../../lib/cookie";
+
+/**
+ * Helper function to get preferences from an external API
+ */
+export async function customGetConsentPreferences(
+  config: FidesConfig
+): Promise<GetPreferencesFnResp | null> {
+  if (!config.options.apiOptions?.getPreferencesFn) {
+    return null;
+  }
+  debugLog(config.options.debug, "Calling custom get preferences fn");
+  try {
+    return await config.options.apiOptions.getPreferencesFn(config);
+  } catch (e) {
+    debugLog(
+      config.options.debug,
+      "Error retrieving preferences from custom API, continuing. Error: ",
+      e
+    );
+    return null;
+  }
+}
+
+/**
+ * Helper function to save preferences to an external API
+ */
+export async function customSaveConsentPreferences(
```

Comment:
these all follow the same pattern so could probably be refactored to share more code, though I know that's tricky because they all take different arguments! would maybe use typescript generics—in any case, not asking you to change it now since it is not trivial, but might be an opportunity for refactoring in the future